src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs

```diff
@@ -598,158 +471,128 @@ public bool TryGetValue(T equalValue, [MaybeNullWhen(false)] out T actualValue)
-                throw new ArgumentNullException(nameof(other));
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.other);
             }
 
-            // intersection of anything with empty set is empty set, so return if count is 0
-            if (_count == 0)
+            // Intersection of anything with empty set is empty set, so return if count is 0.
+            // Same if the set intersecting with itself is the same set.
+            if (Count == 0 || other == this)
             {
                 return;
             }
 
-            // set intersecting with itself is the same set
-            if (other == this)
-            {
-                return;
-            }
-
-            // if other is empty, intersection is empty set; remove all elements and we're done
-            // can only figure this out if implements ICollection<T>. (IEnumerable<T> has no count)
-            ICollection<T>? otherAsCollection = other as ICollection<T>;
-            if (otherAsCollection != null)
+            // If other is known to be empty, intersection is empty set; remove all elements, and we're done.
+            if (other is ICollection<T> otherAsCollection)
             {
                 if (otherAsCollection.Count == 0)
                 {
                     Clear();
                     return;
                 }
 
-                HashSet<T>? otherAsSet = other as HashSet<T>;
-                // faster if other is a hashset using same equality comparer; so check
+                // Faster if other is a hashset using same equality comparer; so check
                 // that other is a hashset using the same equality comparer.
-                if (otherAsSet != null && AreEqualityComparersEqual(this, otherAsSet))
+                if (other is HashSet<T> otherAsSet && AreEqualityComparersEqual(this, otherAsSet))
                 {
-                    IntersectWithHashSetWithSameEC(otherAsSet);
+                    IntersectWithHashSetWithSameComparer(otherAsSet);
                     return;
                 }
             }
 
             IntersectWithEnumerable(other);
         }
 
-        /// <summary>
-        /// Remove items in other from this set. Modifies this set.
-        /// </summary>
-        /// <param name="other">enumerable with items to remove</param>
+        /// <summary>Removes all elements in the specified collection from the current <see cref="HashSet{T}"/> object.</summary>
+        /// <param name="other">The collection to compare to the current <see cref="HashSet{T}"/> object.</param>
         public void ExceptWith(IEnumerable<T> other)
         {
             if (other == null)
             {
-                throw new ArgumentNullException(nameof(other));
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.other);
             }
 
-            // this is already the empty set; return
-            if (_count == 0)
+            // This is already the empty set; return.
```

Comment:
Extreme nit, but I didn't think we generally used periods on comments unless they were followed by more comment.

Reply:
It makes my eye move more slowly but that may well be me.

Reply:
I generally have :)